app/controllers/admin/collections_controller.rb

```diff
@@ -246,9 +246,9 @@ def attach_poster
     authorize! :edit, @collection, message: "You do not have sufficient privileges to add a poster image."
 
     poster_file = params[:admin_collection][:poster]
-    resized_poster = resize_uploaded_poster(poster_file&.path)
-    if resized_poster
-      @collection.poster.content = resized_poster
+    is_image = check_image_compliance(poster_file&.path)
+    if is_image
+      @collection.poster.content = poster_file.tempfile.open.read
```

Comment:
I think this could be simplified:
```suggestion
      @collection.poster.content = poster_file.read
```
https://api.rubyonrails.org/v5.2.3/classes/ActionDispatch/Http/UploadedFile.html#method-i-read